### PR TITLE
Add `unittest.mock` backport

### PR DESCRIPTION
**Pull request by ruaridhw**
## Summary/Motivation:
- To allow usage of `unittest.mock` in packages that import pyutilib.th as unittest

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by jsiirola** [comment]
It looks like the hard dependence on the `mock` library might be causing problems (I rarely use `mock`, and in our other discussion missed that mock was a separate package before Python 3.3).  I am not sure how others will feel about adding another dependency.  At a minimum, we should promulgate the `mock` from `unittest` in Python3:
```python
if six.PY3:
    from unittest import mock
else:
    import mock
```
I wonder if it also makes sense to make the dependency optional:
```python
if six.PY3:
    from unittest import mock
    mock_available = True
else:
    try:
        import mock
        mock_available = True
    except ImportError:
        mock_available = False
```
Thoughts?

**Comment by ruaridhw** [comment]
Updated accordingly though I figured that using Python3 doesn't necessarily imply that `mock` exists in `unittest` since it wasn't around in 3.0

**Comment by jsiirola** [comment]
@ruaridhw: correct, but PyUtilib explicitly doesn't support 3.0-3.3, so we can be lazy.

File: setup.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ def read(*rnames):
                 break
         return line + README.read()
 
-requires=[ 'nose', 'six' ]
+
+requires = ['nose', 'six', 'mock==3.0.5']
```
Comment (jsiirola) [inline setup.py:30]: Please revert the change to the required dependencies.

Comment (ruaridhw) [reply setup.py]: Updated
